### ISSUE TITLE
Plans to support GNOME 40?

### ISSUE THREAD

**Issue by shilangyu** [maoschanz/emoji-selector-for-gnome#151]
Are there plans to support GNOME 40?

**Comment by AlexWayfer** [comment]
Did you see #149?

**Comment by shilangyu** [comment]
I didn't, either way this issue can be used for tracking progress.

**Comment by almereyda** [comment]
This is particularily intruiging, since this package has been selected for the Fedora package repository, and also provides builds for Fedora 34, which is delivered with GNOME 40.

https://src.fedoraproject.org/rpms/gnome-shell-extension-emoji-selector

**Comment by carlwgeorge** [comment]
Howdy, Fedora maintainer here.  I'm happy to update this package as soon as there is a GNOME 40 compatible version tagged here in this repo.

**Comment by vjxyz** [comment]
Please support Gnome 40!  I love this extension and just updated.

**Comment by elektronik2k5** [comment]
Gnome 41 is already out as well.

**Comment by almereyda** [comment]
I'm happily running this on Fedora 34 with GNOME 40 since a while from the
stock extensions repository.

https://extensions.gnome.org/extension/1162/emoji-selector/

What is missing for you here? Is there a specific reason why you cannot
install it from there, or why you have to use the packaged version?

@maoschanz Can you comment on the request by the Fedora package maintainer
above, when a tagged version of the GNOME 40 compatible source code could
become available? Many thanks in advance!

**Comment by Deuchnord** [comment]
I'm not sure how you could install the extension on GNOME 40 from this page: on my side, it has the label _Incompatible_, and the shell version select does not list GNOME 40 (it stops at 3.38).
Are you sure you are running GNOME 40?

**Comment by carlwgeorge** [comment]
Probably `/org/gnome/shell/disable-extension-version-validation` set to true.

**Comment by Bananaman** [comment]
@carlwgeorge I look forward to the new pull request which adds proper 40/41+ support. Should work well in 42 too from there on. :)

I use Fedora by the way, and I wonder what the difference is between this:

https://extensions.gnome.org/extension/1162/emoji-selector/

and this:

https://src.fedoraproject.org/rpms/gnome-shell-extension-emoji-selector

Wouldn't it be better to install it from the extensions web to get updates that way? Hmm?

**Comment by carlwgeorge** [comment]
@Bananaman Are you suggesting that I should send such a pull request?  My expertise is in RPM packaging, not Javascript or CSS, so I'm not the right person for this task.  Volunteering strangers for work you don't want to do yourself isn't really how open source works, so please don't do that in the future.

As far as getting the extension from the extension website or from the Fedora RPM repositories, you get the same code in the end so it's your choice.  Myself and many others prefer to do all installations/updates through the system package manager, which is why I maintain that RPM package.

**Comment by t-lock** [comment]
@carlwgeorge I've been looking for a good first project in the Linux / Gnome ecosystem, and I'm coming from web development, so css/js is my bread and butter. However, I don't know where to begin with extension development... I can look into this on my own, but in the meantime, I have a question that you, @maoschanz or someone else on this thread might be able to answer: 

What I have noticed is that this extension works perfectly with default settings on Gnome 40.4 on Pop OS if I disable extension version validation, but what does not work is the settings. Doesn't this mean that the issue is in the system layer, _not_ something in whatever "web view" powers the tray window?

**Comment by carestad** [comment]
> Probably `/org/gnome/shell/disable-extension-version-validation` set to true.

Setting that allows the extension to be installed successfully, although there could be complications of course :smile:

**Comment by julianferres** [comment]
> > Probably `/org/gnome/shell/disable-extension-version-validation` set to true.
> 
> Setting that allows the extension to be installed successfully, although there could be complications of course smile

It actually installs it and works, but it doesn't have any kind of settings though (just an error message in my case)

![Screenshot from 2022-01-15 04-46-51](https://user-images.githubusercontent.com/32100099/149614130-9c68eb1b-f667-40a2-87fc-933d9f66e330.png)

**Comment by MSandro** [comment]
> > > Probably `/org/gnome/shell/disable-extension-version-validation` set to true.
> > 
> > 
> > Setting that allows the extension to be installed successfully, although there could be complications of course smile
> 
> It actually installs it and works, but it doesn't have any kind of settings though (just an error message in my case)
same in Gnome 41.3

**Comment by Syzygianinfern0** [comment]
Is there no way of getting this to work on Gnome 41? If there is a working fork or anything, could someone please share 🙏🙏🙏

**Comment by joelimgu** [comment]
I haven't tried any forks, but if you add gnome 41 as a supported version of gnome it works. Just settings are broken. But if you have the time to find a working fork please share.

**Comment by Bananaman** [comment]
> Are you suggesting that I should send such a pull request? My expertise is in RPM packaging, not Javascript or CSS, so I'm not the right person for this task. Volunteering strangers for work you don't want to do yourself isn't really how open source works, so please don't do that in the future.

Sorry, I misread your post and thought you were working on a pull reques. :P

**Comment by KarunaGovind** [comment]
@julianferres one of accessing/modifying settings for now can be via gsettings. 

`
# Show all settings
gsettings --schemadir ~/.local/share/gnome-shell/extensions/emoji-selector@maestroschan.fr/schemas list-keys org.gnome.shell.extensions.emoji-selector

# Modify a specific settings
gsettings --schemadir ~/.local/share/gnome-shell/extensions/emoji-selector@maestroschan.fr/schemas get org.gnome.shell.extensions.emoji-selector always-show
true
`

**Comment by Syzygianinfern0** [comment]
> Just to share, I just installed https://github.com/tom-james-watson/Emote - the app size is insane but it works, [I was able to change the shortcut easily](https://github.com/tom-james-watson/Emote/wiki/Hotkey-In-Wayland) ^^" and it pastes instantly the selected emoji. ![image](https://user-images.githubusercontent.com/1692136/165955590-ff803003-da70-46ff-94da-4b4a0f249f78.png)

Does it launch really slowly for you?

**Comment by cmalard** [comment]
[yep :-(](https://github.com/tom-james-watson/Emote/issues/54) switched back to emoji-selector thanks to Karuna's info on how to change the settings without the UI

**Comment by Syzygianinfern0** [comment]
> [yep :-(](https://github.com/tom-james-watson/Emote/issues/54) switched back to emoji-selector thanks to Karuna's info on how to change the settings without the UI

Check out [rofimoji](https://github.com/fdw/rofimoji). It is really snappy and quite customizable.

**Comment by ccoenen** [comment]
> Does it launch really slowly for you?

Yes, when I launched it the first time, i had enough time to recheck if I added the shortcut correctly. Before being presented with three instances (because I'm really impatient). Overall it's better than nothing, but the startup speed is ridiculous.

**Comment by yavko** [comment]
#149 has been merged so this should be closed I think?

**Comment by shilangyu** [comment]
Has it been released? Can someone confirm it works? I'll close the issue then

**Comment by maoschanz** [comment]
the prefs window works but the new version of the extension isn't released yet so the issue should stay opened for now

**Comment by ulville** [comment]
Seems like it's been 19 days since the new version had released on github but it's not on [EGO](https://extensions.gnome.org/extension/1162/emoji-selector/) yet. It's a long time for the review process. Is the new version still waiting for the review? Is there a problem on that side?

**Comment by stephane-klein** [comment]
I have installed master branch (`380b572911b4ff2d8de026cbcefd89f2137143c9`) on Fedora 36 with Gnome `42.3`, `emoji-selector-for-gnome` works with success with Wayland 👍

**Comment by ulville** [comment]
@maoschanz I assume it's rejected on the review. If you can give us a clue why it's been rejected (they usually tell you on the review page, if not you can always ask) I can try to help you for making it meet the requirements of [Review Guidelines](https://gjs.guide/extensions/review-guidelines/review-guidelines.html#gnome-shell-extensions-review-guidelines)

**Comment by maoschanz** [comment]
It's rejected for small things here and there but there is a bigger bug: the recent emojis' list isn't restored when reloading gnome shell, I noticed it only after the rejection 

Sadly I've broken my computer so I can only answer mentions or merge PRs from my phone, I can't even diagnose the issue... (And I can't even develop Drawing!! 😭 my 2+ years weekly contribution streak is ruined 😓)

I've another vacation for the last two weeks of August I'll use that time to install my dear debian Sid on a new machine and finish the gnome 40+ support of this extension, the quicklists one, and the notes one

**Comment by Syzygianinfern0** [comment]
Wow! It works now directly from the gnome extensions store!! :tada:

Great kudos to @maoschanz :heart:

**Comment by ccoenen** [comment]
Thanks for bringing this back :-)

**Comment by t-lock** [comment]
@maoschanz yesssss thank you! Love this package!
